src/rgw/driver/rados/rgw_cr_tools.cc

```diff
@@ -88,7 +88,12 @@ int RGWUserCreateCR::Request::_send_request(const DoutPrefixProvider *dpp)
   }
 
   RGWNullFlusher flusher;
-  return RGWUserAdminOp_User::create(dpp, store, op_state, flusher, null_yield);
+  bool dump_keys = true;
+  int keys_perm = op_state.get_caps_obj()->check_cap("keys", RGW_CAP_READ);
+  if(keys_perm < 0) {
+    dump_keys = false;
+  }
+  return RGWUserAdminOp_User::create(dpp, store, op_state, flusher, dump_keys, null_yield);
```

Comment:
@cbodley I determined the value of `dump_keys`  from the `op_state` here. Was that the right thing to do? I don't know which code paths exercise this `RGWUserAdminOp_User::create()` so I'm not sure if there is an `req_state` somewhere up the call stack.

Reply:
@alimaredia i think it's right, but nothing uses this `RGWUserCreateCR` class after @yuvalif removed pubsub in https://github.com/ceph/ceph/pull/48996. i'd rather remove the class entirely than continue to maintain it